lib/pushy_client/protocol_handler.rb

```diff
@@ -476,6 +468,35 @@ def send_signed_json_command(method, json)
       end
     end
 
+    # This method validate and generate the message
+    # params user packet
+    # Return Waring or JSON message
+    def validate_and_generate_json(json)
```

Comment:
This is generally the shape I was looking for; putting all the size checking and pruning code into one place rather than checking in two different places.